File: Easy/021/rust/src/main.rs
use proconio::input;
use std::collections::VecDeque;

fn generate_prime_list(max_size: u32) -> Vec<u32> {
    let p: u32 = ((max_size + 1) as f32).sqrt().ceil() as u32;
    let capacity: usize = p.pow(2u32) as usize;

    let mut array = VecDeque::<u32>::with_capacity(capacity);
    let mut primes = Vec::<u32>::new();
    for i in 2..capacity+1 { array.push_back(i as u32); }

    loop {
        let value = array.pop_front();
        match value.as_ref() {
            Some(prime) => {
                primes.push(*prime);
                array = array.iter().filter(|&&v| v % *prime != 0).cloned().collect::<VecDeque<_>>();
                if *prime > p {
                    array.iter().for_each(|&v| primes.push(v));
                    break;
                }
            },
            None => { break; }
        }
    }

    return primes;
}

fn main() {
    input! {
        x: u32
    }
    let primes = generate_prime_list(x);
    if primes.len() > 0 {
        for &prime in primes.iter() {
            if prime >= x {
                println!("{}", prime);
                break;
            }
        }
    }
}

File: Easy/021/typescript/src/main.ts
const generate_prime_list = (max_size : number) : number[] => {
  const p : number = Math.ceil(Math.sqrt(max_size + 1));
  const primes : number[] = [];

  const capacity : number = Math.pow(p, 2);
  let array : number[] = [];
  for (let i = 2; i < capacity; i++) {
    array.push(i);
  }

  while (true) {
    let prime : number = array.shift()!;
    primes.push(prime);
    array = array.filter((value) => value % prime !== 0);
    if (prime > p) {
      array.forEach((v) => primes.push(v));
      break;
    }
  }

  return primes;
}


const main = (s: number): number => {
  const primes: number[] = generate_prime_list(s);
  let ret : number = 2;

  for (const prime of primes) {
    if (prime >= s) {
      ret = prime;
      break;
    }
  }

  return ret;
}

(async () => {
  const decoder = new TextDecoder();
  let inputs: string = "";
  for await (const chunk of Deno.stdin.readable) {
    inputs += decoder.decode(chunk).replace(/(\r\n|\n|\r)/gm, " ");
  }

  let s: number = Number(inputs.trim());
  console.log(main(s)); // necessary trimming last space.
})();
